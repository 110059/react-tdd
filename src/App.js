import React from 'react';
import logo from './logo.svg';
import './App.css';
import Header from './components/Header';
import Footer from './components/Footer';
import CheckboxWithLabel from './components/CheckboxWithLabel';
import Button from './components/Button';
import AppNew from './components/AppNew';

function App() {
  return (
    <div >
       {/* <Header />
        <Footer />
        <CheckboxWithLabel />
        <Button /> */}
        <AppNew />
    </div>
  );
}

export default App;
