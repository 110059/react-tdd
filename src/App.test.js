import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

describe("test app", () => {
    test("App componet render no child", () => {
        shallow(<App />);
    });
    test("App componet render with child component", () => {
        mount(<App />);
    });

    test('snapshot renders', () => {
        const component = renderer.create(<App />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

