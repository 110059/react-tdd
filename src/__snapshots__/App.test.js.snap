// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app snapshot renders 1`] = `
<div>
  <div>
    <h1>
      My Counter
    </h1>
    <div>
      <p>
        0
      </p>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      Increment
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Decrement
    </button>
    <h2>
      My Async Data
    </h2>
    <ul />
  </div>
</div>
`;
